### PR TITLE
fix(git): remove default gitTimeout

### LINKED ISSUE #15179: Just started hitting git timeout on our mono repo
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Always the latest. 

### Please select which platform you are using if self-hosting.

Azure DevOps Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

Azure DevOps

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

As of Monday (April 18th)(latest RenovateBot version), we started seeing git clone timeouts issues. 

It appears like some git timeout functionality was added things started to break. Or at least I assume it's related to that. Our git repo takes approximately 5-10 minutes to clone.
Unless I am misunderstanding the gitTimeout mechanism. I just know we started seeing timeout issues around git as of Monday.

I've tried running 10+ times and it consistently is hitting hit 100% of the time. Other smaller repos are still running ifne. 

```
DEBUG: Performing blobless clone (repository=Xbox/OurRepo)
DEBUG: Git function thrown (repository=Xbox/OurRepo)
...
         "plugin": "timeout",
         "message": "block timeout reached",
```

Is there a way to disable the git timer for our case? Or any suggestions on what should be changed as we typically control any timeouts on the pipeline/task level instead.

I know we can pin to an older version of RenovateBot to temporarily workaround this, but would prefer to find a suitable long term solution.

### Relevant debug logs

<details><summary>Logs</summary>

DEBUG: Performing blobless clone (repository=Xbox/GS)
DEBUG: Git function thrown (repository=Xbox/GS)
       "err": {
         "task": {
           "commands": [
             "clone",
             "--filter=blob:none",
             "-c",
             "http.extraheader=AUTHORIZATION: bearer **redacted**",
             "https://microsoft.visualstudio.com/DefaultCollection/Xbox/_git/GS",
             "."
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "plugin": "timeout",
         "message": "block timeout reached",
         "stack": "Error: block timeout reached\n    at Timeout.kill [as _onTimeout] (C:\\npm\\prefix\\node_modules\\renovate\\node_modules\\simple-git\\dist\\cjs\\index.js:1327:24)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)"
       }

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Sets default `gitTimeout` value to `0`.

## Context

Closes #15179, Closes #15177

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/validation.ts [hunk 1]
```diff
@@ -128,19 +128,6 @@ export async function validateConfig(
       });
       continue;
     }
-    if (key === 'gitTimeout') {
-      if (!is.number(val)) {
```
Comment (viceice) [inline lib/config/validation.ts:132]: shouldn't we leave the number check?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/config/validation.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/validation.ts:132]: I thought we automatically check such types. I'll verify manually

Comment (viceice) [reply lib/config/validation.ts:132]: yeah, we should validate by default. 🙃

Comment (rarkins) [reply lib/config/validation.ts:132]: https://github.com/renovatebot/renovate/pull/15206

**Review by viceice — APPROVED** [review]

**Pull request merged**